### PR TITLE
feat(core): support ts 4.8

### PR DESCRIPTION
**Pull request by Coly010**
Add TS 4.8 support

### CODE REVIEW

File: e2e/angular-core/src/angular-linting.test.ts [hunk 1]
```diff
@@ -64,7 +64,6 @@ describe('Angular Package', () => {
       expect(appLintStdOut).toContain(
         `The selector should start with one of these prefixes`
       );
-      expect(appLintStdOut).toContain(`7:18`);
```
Comment (leosvelperez) [inline e2e/angular-core/src/angular-linting.test.ts:67]: Why is this assertion removed?

File: packages/angular/migrations.json [hunk 2]
```diff
@@ -1514,6 +1514,19 @@
           "alwaysAddToPackageJson": false
         }
       }
+    },
+    "14.7.0": {
+      "version": "14.7.0-beta.0",
```
Comment (leosvelperez) [inline packages/angular/migrations.json:1519]: There's a `14.7.0-beta.1` version already released. Let's bump the target version here and in the other `migrations.json` files included in this PR.

File: packages/nx/src/utils/versions.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
 export const nxVersion = require('../../package.json').version;
 
 export const angularCliVersion = '~14.2.0';
-export const typescriptVersion = '~4.7.2';
+export const typescriptVersion = '~4.8.2';
 export const prettierVersion = '^2.6.2';
 export const tslintVersion = '~6.1.0';
-export const typescriptESLintVersion = '~5.33.1';
+export const typescriptESLintVersion = '^5.33.1';
```
Comment (leosvelperez) [inline packages/nx/src/utils/versions.ts:7]:
```suggestion
export const typescriptESLintVersion = '^5.36.1';
```

**Review by leosvelperez — CHANGES_REQUESTED** [review]

File: e2e/angular-core/src/angular-linting.test.ts (hunk 1 above)
Comment (Coly010) [reply e2e/angular-core/src/angular-linting.test.ts:67]: I may be wrong, but I believe this is checking the exact line number that the error occurred on, or perhaps the error number? Either way, this part of the error message changed, but the error itself (the text above) did not. If this is subject to change across versions of typescript-eslint then I'd rather not rely on it, as it does not provide much value.

File: packages/angular/migrations.json (hunk 2 above)
Comment (Coly010) [reply packages/angular/migrations.json:1519]: good spot

File: packages/nx/src/utils/versions.ts (hunk 3 above)
Comment (Coly010) [reply packages/nx/src/utils/versions.ts:7]: :+1:

**Review by JamesHenry — CHANGES_REQUESTED** [review]
See below

**Review by JamesHenry — DISMISSED** [review]
See below

**Review by JamesHenry — COMMENTED** [review]
Feedback reapplied as non-blocking suggestion:

---

This change will cause Nx to only work with TS4.8 because the getDecorators and getModifiers functions did not exist before then. It is easy to work around and therefore IMO it would be worth doing that and aiming to not lose support prior to 4.8 in order to cut down on issues reported in a minor version bump when users inevitably don't migrate their workspaces as we would hope.

This is how we wrapped it in typescript-eslint

https://github.com/typescript-eslint/typescript-eslint/blob/a603015c93a6ea96d500c57bec1e284340141b1f/packages/typescript-estree/src/getModifiers.ts

You could potentially use the @typescript-eslint/type-utils package and leverage these utils directly.

**Review by leosvelperez — CHANGES_REQUESTED** [review]

File: packages/angular/src/generators/utils/insert-ngmodule-import.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { applyChangesToString, ChangeType, Tree } from '@nrwl/devkit';
 import {
   __String,
   CallExpression,
+  canHaveDecorators,
```
[the remark this answers is not in the document]
Comment (Coly010) [reply packages/angular/src/generators/utils/insert-ngmodule-import.ts]: Good spot, resolved now

**Review by leosvelperez — APPROVED** [review]
LGTM! 🎉

**Review by JamesHenry — COMMENTED** [review]
LGTM! The only thing I might suggest is adding a comment by the imports to explain why we are pulling those utils in (ie not sacrificing TS < 4.8 support), as it might not be obvious to contributors in future who weren’t involved in this PR

**Comment by Coly010** [comment]
> LGTM! The only thing I might suggest is adding a comment by the imports to explain why we are pulling those utils in (ie not sacrificing TS < 4.8 support), as it might not be obvious to contributors in future who weren’t involved in this PR

Very good point

**Review by JamesHenry — APPROVED** [review]

**Pull request merged**